orbit/pkg/update/notifications.go

```diff
@@ -83,6 +88,22 @@ func (h *renewEnrollmentProfileConfigFetcher) GetConfig() (*fleet.OrbitConfig, e
 					return cfg, nil
 				}
 
+				// we perform this check locally on the client too to avoid showing the
+				// dialog if the Fleet enrollment profile has not been assigned to the device in
+				// Apple Business Manager.
+				assignedFn := h.checkAssignedEnrollmentProfileFn
+				if assignedFn == nil {
+					assignedFn = profiles.CheckAssignedEnrollmentProfile
+				}
+				if err := assignedFn(h.fleetURL); err != nil {
+					log.Error().Err(err).Msg("checking assigned enrollment profile")
+					log.Info().Msg("a request to renew the enrollment profile was processed but not executed because there was an error checking the assigned enrollment profile.")
+					// TODO: Design a better way to backoff `profiles show` so that the device doesn't get rate
+					// limited by Apple. For now, wait at least 1 minute before retrying.
+					h.lastRun = time.Now().Add(-h.Frequency).Add(1 * time.Minute)
```

Comment:
I also just realized that this also depends on `h.Frequency`, let's double check that value and verify that's working as expected (seems to be 1 hour, but we got reports that runs way more frequently?)

Reply:
Frequency is 1 hour, but it appears to not be working as expected because the `lastRun` isn't being incremented when the `renew` command fails with an error, which means it tries again immediately the next time the config fetcher loops back around. I added a note where we might want to address that below.

Reply:
I experimented with a few different approaches to the backoff and ultimately decided to just up the wait to two minutes and keep it simple for now.  I figure we'll probably need something to handle backoff on the other commands too and didn't want to spend too much time designing that now, especially as we're still trying to understand how Apple is actually implementing the rate-limiting.

Reply:
but even then, the customer reports seeing the notification multiple times in a row. Is the notification showing up but we get a non-zero exit code? is there a bug hidden somewhere? we should look into that, but agreed that separately.

Reply:
> 

I experimented with a few different approaches to the backoff and ultimately decided to just up the wait to two minutes and keep it simple for now. I figure we'll probably need something to handle backoff on the other commands too and didn't want to spend too much time designing that now, especially as we're still trying to understand how Apple is actually implementing the rate-limiting.

that sounds like a plan, will review in more detail tomorrow, but looks good at first glance